apps/web/pages/event-types/index.tsx

```diff
@@ -55,293 +36,6 @@ const CreateFirstEventTypeView = ({ canAddEvents, profiles }: CreateEventTypePro
-                                    showToast(t("link_copied"), "success");
-                                  }}
-                                  className={classNames(
-                                    active ? "bg-neutral-100 text-neutral-900" : "text-neutral-700",
-                                    "group flex w-full items-center px-4 py-2 text-sm font-medium"
-                                  )}>
-                                  <ClipboardCopyIcon
-                                    className="mr-3 h-4 w-4 text-neutral-400 group-hover:text-neutral-500"
-                                    aria-hidden="true"
-                                  />
-                                  {t("copy_link")}
-                                </button>
-                              )}
-                            </Menu.Item>
-                            {isNativeShare ? (
-                              <Menu.Item>
-                                {({ active }) => (
-                                  <button
-                                    onClick={() => {
-                                      navigator
-                                        .share({
-                                          title: t("share"),
-                                          text: t("share_event"),
-                                          url: `${process.env.NEXT_PUBLIC_APP_URL}/${profile.slug}/${type.slug}`,
-                                        })
-                                        .then(() => showToast(t("link_shared"), "success"))
-                                        .catch(() => showToast(t("failed"), "error"));
-                                    }}
-                                    className={classNames(
-                                      active ? "bg-neutral-100 text-neutral-900" : "text-neutral-700",
-                                      "group flex w-full items-center px-4 py-2 text-sm font-medium"
-                                    )}>
-                                    <UploadIcon
-                                      className="mr-3 h-4 w-4 text-neutral-400 group-hover:text-neutral-500"
-                                      aria-hidden="true"
-                                    />
-                                    {t("share")}
-                                  </button>
-                                )}
-                              </Menu.Item>
-                            ) : null}
-                          </div>
-                        </Menu.Items>
-                      </Transition>
-                    </>
-                  )}
-                </Menu>
-              </div>
-            </div>
-          </li>
-        ))}
-      </ul>
-    </div>
-  );
-};
-
-interface EventTypeListHeadingProps {
-  profile: EventTypeGroupProfile;
-  membershipCount: number;
-}
-const EventTypeListHeading = ({ profile, membershipCount }: EventTypeListHeadingProps): JSX.Element => (
```

Comment:
I'm assuming this got extracted to a an external file. We kinda have an unspoken rule about separating components. We only do it once it is needed in more than one place. If it's not, let's keep it in the same file for now.

Reply:
I think I'm not following. How does separating this file will help with scaling? Is it going to be used into multiple files? If that's the case. Let's separate it when that happens. Not before. This will generate confusion since we will assume that's is being used in many places when it's not. We rather have everything in just one file if it's not needed anywhere else.

Reply:
More context for this decision: https://github.com/calcom/cal.com/pull/923#pullrequestreview-777931318

Reply:
Loud and clear. Reverted the smaller components. I guess it would be a good idea to gather those little pieces of advise and technical decisions and put them together in a best practices page. Or perhaps that is already there and I missed it 😅